cmd/freeze-checkpoint/app/root.go

```diff
@@ -0,0 +1,232 @@
+//
+// Copyright 2025 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+
+	gcs "cloud.google.com/go/storage"
+	rekornote "github.com/sigstore/rekor-tiles/pkg/note"
+	"github.com/sigstore/rekor-tiles/pkg/signerverifier"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	logformat "github.com/transparency-dev/formats/log"
+	"github.com/transparency-dev/trillian-tessera/api/layout"
+	"golang.org/x/mod/sumdb/note"
+)
+
+const (
+	frozenString = "Log frozen — "
```

Comment:
@haydentherapper before I merge this do you have any thoughts on using this string + date to indicate the frozen log? A monitor is going to have to be checking for this string so human readability isn't the only consideration.

Reply:
This looks good, monitors can easily check for this constant string. I'm not sure if the timestamp will be necessary since we will be co-signing the checkpoint and getting signed timestamps later on, but it's helpful now and would be fine in the future too.